### PR TITLE
Moved several consensus fields to state struct.

### PR DESCRIPTION
**Pull request by Frozen**
Motivation:
Refactored the consensus module by splitting it into smaller, modular structs to enhance testability and maintainability.

Results:

- Increased test coverage for consensus statements from 15% to 18%.
- Successfully addressed and validated corner cases through comprehensive testing.

### CODE REVIEW

**Comment by sophoah** [comment]
@GheisMohammadi please review

File: consensus/state.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import (
+	"sync/atomic"
+	"unsafe"
+
+	bls_cosi "github.com/harmony-one/harmony/crypto/bls"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/rs/zerolog"
+)
+
+// State contains current mode and current viewID
```
Comment (sophoah) [inline consensus/state.go:12]: this was copied from view_change.go, please update description accordingly

**Review by GheisMohammadi — APPROVED** [review]

**Review by sophoah — APPROVED** [review]
